apps/admin/backend/src/adjudication.ts

```diff
@@ -0,0 +1,110 @@
+import { assert, assertDefined, throwIllegalValue } from '@votingworks/basics';
+import { VoteAdjudication, WriteInAdjudicationAction } from './types';
+import { type Store } from './store';
+
+/**
+ * Manipulates adjudication records so that a particular vote in a cast vote
+ * record reflects the target marked or unmarked status. Ensures that
+ * adjudications are not created when the scanned vote is already the target
+ * status.
+ */
+export function adjudicateVote(
+  voteAdjudication: Omit<VoteAdjudication, 'id'>,
+  store: Store
+): void {
+  const { votes } = store.getCastVoteRecordVoteInfo({
+    electionId: voteAdjudication.electionId,
+    cvrId: voteAdjudication.cvrId,
+  });
+
+  const contestVotes = votes[voteAdjudication.contestId];
+
+  const isScannedVote = contestVotes
+    ? contestVotes.includes(voteAdjudication.optionId)
+    : /* c8 ignore next 1 */
+      false;
```

Comment:
yeah, this should be replaced by an assertion, thank you. It actually wasn't possible before #4195 either, as the votes in VxAdmin represent the votes in the CVR, which include the empty arrays.

Reply:
Revising my response to be the same as for a similar one above?

It shouldn't be possible, but the testing infrastructure I have for mocking specific cast vote records does not have this guarantee. Because our fixtures don't have gridLayouts, I don't have a good means to make those mocks accurate, so leaving for now.